test/integration_test.py

```diff
@@ -350,11 +352,10 @@ def _prune_test_bundles(self, bundle_fqids, max_bundles):
 
     def _queue_empty_timeout(self, num_bundles: int):
         time_per_bundle = 5
-        time_for_deque_to_empty = 10 * 5  # time for a 0 count to propagate through the deque in wait_for_queue_level()
-        min_timeout = 60
-        max_timeout = 10 * 60
-        timeout = max((num_bundles * time_per_bundle) + time_for_deque_to_empty, min_timeout)
-        return min(timeout, max_timeout)
+        deque_size = 10
```

Comment:
This reveals an ugly brittle coupling. While this is not Please eliminate this method and the `timeout` argument to `wait_for_queue_level`. Replace it with a `num_bundles` argument that defaults to `None` (meaning "assume many bundles"). In `wait_for_queue_level`, derive the timeout from that `num_bundles` argument.

Reply:
It's only fair that this is fixed in this PR because the coupling appears to have been introduced in https://github.com/DataBiosphere/azul/commit/87aff4ddb02e989d603dda7ca87c0b774caff638.